### PR TITLE
coscheduling: cleanup and fixes

### PR DESCRIPTION
**Pull request by yuanchen8911**
Made various edits and minor fixes to the coscheduling code. 

1. Added/updated comments.
2. Fixed typos and grammar glitches.
3. Renamed a couple of functions/variables.

Remaining issues. 
1. Consistent logging levels for klog.
2. Make DenyCache expiration time configurable.          
3. Remove the outdated parameters for the PodGroup label version.

### CODE REVIEW

**Review by Huang-Wei — COMMENTED** [review]
Some nits. LGTM otherwise.

**Comment by denkensk** [comment]
Thank you for your improvement.
/lgtm

File: pkg/coscheduling/core/core.go [hunk 1]
```diff
@@ -119,21 +119,21 @@ func (pgMgr *PodGroupManager) Permit(ctx context.Context, pod *corev1.Pod, nodeN
...
+	ready := int32(assigned)+1 >= pg.Spec.MinMember
 	if ready {
 		return true, nil
 	}
 	return false, util.ErrorWaiting
 }
 
-// PostBind is used for send metrics and update to api if all finish
+// PostBind updates a PodGroup's status.
 func (pgMgr *PodGroupManager) PostBind(ctx context.Context, pod *corev1.Pod, nodeName string) {
```
[the remark this answers is not in the document]
Comment (Huang-Wei) [reply pkg/coscheduling/core/core.go:137]: Don't want this to block merging. @cwdsuzhou we can discuss it later, maybe merge it to your PR addressing PreScheduling concern, or a separate PR.

[the remark this answers is not in the document]
Comment (cwdsuzhou) [reply pkg/coscheduling/core/core.go:137]: IMO,  keep this is also ok.  Only update status to apiserver when phase changed, so I think we would not create many requests there.

**Pull request merged**